frontend/pages/SoftwarePage/SoftwareTitleDetailsPage/SoftwarePackageCard/SoftwarePackageCard.tsx

```diff
@@ -263,7 +264,17 @@ const SoftwarePackageCard = ({
           `Byte size (${resp.data.size}) does not match content-length header (${contentLength})`
         );
       }
-      const filename = name;
+
+      let filename = name;
+      try {
+        const cd = parse(resp.headers["content-disposition"]);
+        if (cd.parameters.filename) {
+          filename = cd.parameters.filename;
+        }
+      } catch (e) {
+        // noop, we'll just use the name as is
```

Comment:
it's really a stretch, so don't worry about it. But any chance we have the platform of the title to add the extension using? https://github.com/fleetdm/fleet/blob/92ccf08dd245d35d662cee282931f4a6048e9438/frontend/interfaces/installer.ts#L9-L17

Reply:
Sure, I can see about adding that

Reply:
not super important, so don't worry too much about it

Reply:
Ok, I was hoping we had access to the full title object in this component, but unfortunately it would require some refactoring of the component props so I'll leave it as a todo.

Reply:
makes sense, thanks for trying!